lib/features/category/presentation/widgets/category_products.dart

```diff
@@ -1,35 +1,59 @@
 import 'package:flutter/material.dart';
 import 'package:shopping_app/core/models/category_model.dart';
 import 'package:shopping_app/core/presentation/widgets/products_box.dart';
-import 'package:shopping_app/features/category/presentation/view_model/category_view_model.dart';
+import 'package:shopping_app/features/category/presentation/view_model/category_products_viewmodel.dart';
 import 'package:shopping_app/features/category/presentation/widgets/items_loading_listview.dart';
 
 class CategoryProducts extends StatefulWidget {
   final Category category;
-  const CategoryProducts({super.key, required this.category});
+  final ValueNotifier<String> searchQuery;
+  const CategoryProducts({
+    super.key,
+    required this.category,
+    required this.searchQuery,
+  });
 
   @override
   State<CategoryProducts> createState() => _CategoryProductsState();
 }
 
 class _CategoryProductsState extends State<CategoryProducts> {
-  final CategoryViewModel categoryViewModel = CategoryViewModel();
+  final CategoryProductsViewModel categoryProductsViewModel =
+      CategoryProductsViewModel();
+
+  void _searchQueryListener() {
+    categoryProductsViewModel.search(widget.searchQuery.value);
+  }
 
   @override
   void initState() {
-    categoryViewModel.getProductsInCategory(widget.category);
     super.initState();
+    categoryProductsViewModel.getProductsInCategory(widget.category).then(
```

Comment:
@UI-Light the fix to get search to work when you switch tabs is here.
Because every time you switch tabs, getProductsInCategory is called again for that specific tab, your search is not useful until that call to get the products is completed.

So I have used a then callback on that future to perform the search after it has fetched the products.

A better solution would be to prevent the tabs from loading products every single time you switch from one tab to the other. But this is a good fix for now.

Let me know if you understand